### ISSUE TITLE
Pull windmill to main is always failing

### ISSUE THREAD

**Issue by pa-bru** [windmill-labs/windmill-sync-example#6]
I want to implement this workflow but I noticed `pull-workspace.yaml` fails mainly during the `pull changes` step. 

here is a screenshot of the error

<img width="1272" alt="Screenshot 2023-05-24 at 14 45 00" src="https://github.com/windmill-labs/windmill-sync-example/assets/14804459/008c4016-1909-41bd-bb14-2f9f608b87f4">

**Comment by rubenfiszel** [comment]
https://github.com/nestdotland/nest.land/issues/336 nest land seems to be down. I will see how we can fix our dependencies

**Comment by rubenfiszel** [comment]
I purged the CLi from this faulty registry and updated the example to use 1.104.0. Thanks for the report

**Comment by pa-bru** [comment]
Thank you @rubenfiszel for the reactivity! I still encounter an issue but maybe I do something wrong. 
For now my repo has nothing in it. everything (scripts, flows, etc) are in the remote workspace. And I'm using a self hosted windmill application. 

it fails here: https://github.com/windmill-labs/windmill/blob/main/cli/pull.ts#LL33C2-L33C2

Do you have any idea? I tried to export the workspace as a zip in the app and it worked well. 

<img width="1204" alt="Screenshot 2023-05-24 at 17 37 24" src="https://github.com/windmill-labs/windmill-sync-example/assets/14804459/87d81b9b-c7a5-4c02-97f5-68c93c6a5970">

**Comment by pa-bru** [comment]
You are right. I just tested the request and it returns html. Our windmill app is hosted on cloudflare and we have to signin with google workspace...

**Comment by pa-bru** [comment]
Do you refer to this [feature](https://github.com/windmill-labs/windmill#oauth-for-self-hosting)?
I also saw the [google cloud github action](https://github.com/marketplace/actions/authenticate-to-google-cloud) so I wonder what I should use

**Comment by rubenfiszel** [comment]
Yes I was referring to configuring oauth so that google sso was a login provider. A possibility is also to self-host the github runner behind your cloudflare tunnet so that it can talk to windmill directly.
